File: code/06/project06/Array.rb
# array.rb prompts for values to go in array and prints them
# Precondition: values are doubles
# Output: values of the array
# Project06 CS 214, Calvin College
# By: Mark Davis
# Date: March 29, 2016
########################################################

###############################################
# readArray() fills array with user's value
# Receive: anArray, size of array
################################################
def readArray(array, size)
	for i in 0 .. size-1 #loop to iterate through array
		print "Enter value: "
		value = gets.chomp.to_i #convert to integer
		array[i] = value #store value in array
	end
	
end


###############################################
# printArray() prints elements of the array
# Receive: anArray, array's size
################################################
def printArray(array, size)
	print "Values are: "
	for i in 0 .. size-1 #loop to iterate through array
		puts array[i] #output element in array
	end
end


def main
	print "Enter size of the array: " #prompt for input
	size = gets.chomp.to_i #store input
	array = Array.new(size) #declare array
	readArray(array, size) #call read
	printArray(array, size) #call print
end

main

File: code/06/project06/Array.java
/* array.java prompts for values and stores them in an array
 * By: Mark Davis
 * Date: March 29, 2016
 * Precondition: the values are numbers
 * Output: the array's values
 ***************************************************************/
import java.util.Scanner; //include class for Scanner

public class Array {
	public static void main(String[] args) {
		Array ar = new Array(); //create instance of class
		System.out.println("Enter the size of the array: " ); //prompt for user input
		Scanner keyboard = new Scanner(System.in); //scan keyboard for input
		int size = keyboard.nextInt(); //store input as array size
		double[] theArray = new double[size]; //declare array
		ar.readArray(theArray, size); //call read function
		ar.printArray(theArray, size); //call print function
		}
	
	/* readArray() prompts user for values and stores them in the array
   	* Receive: anArray, size of the array
   	* Precondition: inputs are doubles
   	************************************************************/
	void readArray(double theArray[], int size) {
		Scanner keyboard = new Scanner(System.in); // scan keyboard for input
		for (int i = 0; i < size; i++) { //loop to iterate through array
			System.out.println("Enter the value for element " + (i+1)); //prompt for input
			double value = keyboard.nextDouble(); // store input
			theArray[i] = value; //store input in array
		}
	}

	/* printArray() outputs all elements in array
   	* Receive: anArray, size of the array
   	* Precondition: parameters are doubles and ints
   	************************************************************/
	void printArray(double theArray[], int size) {
		for (int i = 0; i < size; i++) { //loop to iterate through array
			System.out.println("The values of the array are: " + theArray[i]); //output elements of array
		}
	}
}
